torch/lib/THD/master_worker/common/CommandChannel.cpp

```diff
@@ -53,6 +53,7 @@ MasterCommandChannel::MasterCommandChannel()
   std::tie(_port, world_size) = load_master_env();
 
   _sockets.assign(world_size, -1);
+  _mutexes = std::vector<std::mutex>(world_size);
```

Comment:
It will not compile for sure. I do not think we can avoid using unique_ptr/shared_ptr here.